tests/test_downloadermiddleware_httpcompression.py

```diff
@@ -27,8 +27,11 @@
 class HttpCompressionTest(TestCase):
 
     def setUp(self):
-        self.spider = Spider('foo')
-        self.mw = HttpCompressionMiddleware()
+        #need to find way to access settings here
+        crawler = get_crawler(Spider,{'HTTPCOMPRESSION_HEADERS_KEEP':True,
+                                      'COMPRESSION_ENABLED':True})
+        self.spider = crawler._create_spider('foo')
+        self.mw = HttpCompressionMiddleware.from_crawler(crawler)
```

Comment:
Maybe this should be moved out of the setUp, extracted into a separate function that returns a spider and a middleware, and received an optional parameter that determines whether `HTTPCOMPRESSION_HEADERS_KEEP` is enabled or not. This function can then be called at the beginning of every test with the right parameters for each test.

Reply:
I can understand that. I just modeled this off of the ajax downloadermiddleware test did in its setup function.